// @applitools/eyes-api/dist/Logger.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Logger = void 0;
const utils = __importStar(require("@applitools/utils"));
const logger_1 = require("@applitools/logger");
const LogHandler_1 = require("./input/LogHandler");
class Logger {
    constructor(loggerOrOptionsOrShow = false) {
        /** @internal */
        this.isLogger = true;
        if (utils.types.isBoolean(loggerOrOptionsOrShow)) {
            return new Logger({ show: loggerOrOptionsOrShow });
        }
        else if (utils.types.has(loggerOrOptionsOrShow, ['log', 'console'])) {
            this._logger = loggerOrOptionsOrShow;
        }
        else {
            this._show = loggerOrOptionsOrShow.show;
            this._label = loggerOrOptionsOrShow.label;
            this._handler = loggerOrOptionsOrShow.handler;
        }
    }
    _makeLogger() {
        return (0, logger_1.makeLogger)({
            handler: this._handler instanceof LogHandler_1.LogHandlerData ? this._handler.toJSON() : this._handler,
            level: this._show || utils.general.getEnvValue('SHOW_LOGS', 'boolean') ? 'info' : 'silent',
            label: this._label,
        });
    }
    /** @internal */
    getLogger() {
        if (!this._logger)
            this._logger = this._makeLogger();
        return this._logger;
    }
    getLogHandler() {
        if (!this._handler) {
            return new LogHandler_1.NullLogHandlerData();
        }
        else if (!utils.types.has(this._handler, 'type')) {
            return this._handler;
        }
        else if (this._handler.type === 'file') {
            return new LogHandler_1.FileLogHandlerData(true, this._handler.filename, this._handler.append);
        }
        else if (this._handler.type === 'console') {
            return new LogHandler_1.ConsoleLogHandlerData(true);
        }
    }
    setLogHandler(handler) {
        this._show = true;
        this._handler = handler;
    }
    verbose(...messages) {
        if (!this._logger)
            this._logger = this._makeLogger();
        messages.forEach(message => this._logger.log(message));
    }
    log(...messages) {
        if (!this._logger)
            this._logger = this._makeLogger();
        messages.forEach(message => this._logger.log(message));
    }
    warn(...messages) {
        if (!this._logger)
            this._logger = this._makeLogger();
        messages.forEach(message => this._logger.warn(message));
    }
    error(...messages) {
        if (!this._logger)
            this._logger = this._makeLogger();
        messages.forEach(message => this._logger.error(message));
    }
    fatal(...messages) {
        if (!this._logger)
            this._logger = this._makeLogger();
        messages.forEach(message => this._logger.fatal(message));
    }
    open() {
        this._logger.open();
    }
    close() {
        this._logger.open();
    }
    tag(name, value) {
        this._logger.tag(name, value);
    }
    extend(optionsOrLabel) {
        var _a;
        if (utils.types.isString(optionsOrLabel))
            return this.extend({ label: optionsOrLabel });
        if (this._logger)
            return new Logger(this._logger.extend(optionsOrLabel));
        return new Logger({ show: this._show, label: (_a = optionsOrLabel === null || optionsOrLabel === void 0 ? void 0 : optionsOrLabel.label) !== null && _a !== void 0 ? _a : this._label, handler: this._handler });
    }
}
exports.Logger = Logger;
